_posts/2017-05-08-refactoring-identities.md

```diff
@@ -0,0 +1,50 @@
+
+Meanwhile, for people who actually wanted to manage multiple personas -- pseudonyms, brand identities, etc. -- we never managed to provide a good experience. It was very easy to accidentally use the wrong identity and reveal yourself. At the same time, the need to support this use case often exacerbated the confusion for users who only had one logical persona (as described in the previous paragraph). Without this design constraint, we could make a better UX for most people.
+
+Finally, in practice this design didn't really solve the ability to move grains between Sandstorm servers, because different Sandstorm servers may very well use totally different login mechanisms. Often, people want to transition grains from Oasis -- where they used e-mail login -- to a new self-hosted server -- where they used LDAP or SAML login. Since the login mechanisms differed, user IDs didn't match up anyway. To really solve this problem, we need something more dynamic.
+
+### What we're doing instead
+
+We are dropping support for multiple "personas". Going forward, each Sandstorm account will have only one profile -- one name, one profile picture, one entry in the sharing auto-complete list, etc. What we call "identities" today will be renamed to "credentials". A credential will no longer have its own profile, and you will no longer need to choose which credential you are acting as when accessing grains. Authentication-related features of credentials (for login, and for secure sharing) will remain mostly intact.
+
+Users who rely on the ability to manage multiple personas today will need to transition to using multiple accounts instead. I suspect that there are vanishingly few such users, as most users never understood Sandstorm's identity system in the first place. That said, if you are affected, I apologize. I would love it if you would get in contact with us to let us know about your specific needs, so that we can try to design a better experience for you. (For what it's worth, I personally recommend using the multi-profile feature provided by various browsers to separate your personas into totally separate browser contexts with different window themes -- this makes it much easier to prevent accidental leakage.)
```

Comment:
- It's unclear what the impact will be on existing users with multiple personas.

- Provide the best way to get in touch with us: support@sandstorm.io/dev group/etc.